apple/internal/testing/apple_test_rule_support.bzl

```diff
@@ -115,6 +115,7 @@ def _get_template_substitutions(test_type, test_bundle, test_environment, test_h
     subs["test_type"] = test_type.upper()
     subs["test_env"] = ",".join([k + "=" + v for (k, v) in test_environment.items()])
     subs["test_filter"] = test_filter or ""
+    subs["product_module_name"] = product_module_name or ""
```

Comment:
Doesn't the `test_filter` address each test by the `PRODUCT_MODULE_NAME`?  Perhaps can you parse this from `test_filter` if it it exists. The thing I liked about this is being able to filter different module names. Say you've got a number of modules in the test:

```
bazel test :MyTest --test_filters MyFirst.swiftmodule:foo
bazel test :MyTest --test_filters MySecond.swiftmodule:bar
```

Then - inside of `xctestrunner` - you'd simply set `PRODUCT_MODULE` in the xctestrun based on the filter - and be sure to get the right one. Probably validate they set that.. 

The downside of adding this at the rule level is you can't filter dynamically like that. 

PS : I wonder if this is a regression - or perhaps `test_filter` only ever workd for objc!

Reply:
Also do you imagine that we'd ever want to feed `.xctestrun` ( or partials ) into this rule? That might be another thing to contend with - similar our plist composition stuff in rules_ios you might be able to pass overrides.

I would still question if you can leverage the build graph too: assuming that you pack ( filterable ) tests into the first `.swiftmodule` file - you can set the `PRODUCT_MODULE_NAME` based on the first direct `SwiftInfo`

Obviously these are 2 different solutions. I liked the dep graph way - and might be a nice default - it has some inflexibility but it's probably easier for 99% of tests out there

Reply:
Ah cool glad it works - this makes Bazel easier to use in general IMO